File: src/main/webapp/app/entities/reports/services/sales-summary-report.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient, HttpResponse, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

import { SERVER_API_URL } from 'app/app.constants';
import { IDailySalesSummary } from 'app/shared/model/report/daily-sales-summary.model';

type EntityArrayResponseType = HttpResponse<IDailySalesSummary[]>;

@Injectable({ providedIn: 'root' })
export class SalesSummaryReportService {
  private readonly resourceUrl = SERVER_API_URL + 'api/sales-summary';
  private readonly http = inject(HttpClient);

  /**
   * Get daily sales summary for a date range
   */
  getDailySalesSummary(startDate: string, endDate: string): Observable<EntityArrayResponseType> {
    const params = new HttpParams().set('startDate', startDate).set('endDate', endDate);
    return this.http.get<IDailySalesSummary[]>(this.resourceUrl, { params, observe: 'response' });
  }

  /**
   * Get daily sales summary for a specific date
   */
  getDailySalesSummaryByDate(date: string): Observable<EntityArrayResponseType> {
    const params = new HttpParams().set('date', date);
    return this.http.get<IDailySalesSummary[]>(`${this.resourceUrl}/by-date`, { params, observe: 'response' });
  }

  /**
   * Get daily sales summary filtered by sale type
   */
  getDailySalesSummaryByType(startDate: string, endDate: string, typeVente: string): Observable<EntityArrayResponseType> {
    const params = new HttpParams().set('startDate', startDate).set('endDate', endDate).set('typeVente', typeVente);
    return this.http.get<IDailySalesSummary[]>(`${this.resourceUrl}/by-type`, { params, observe: 'response' });
  }
}

File: src/main/java/com/kobe/warehouse/service/report/SalesSummaryReportService.java
package com.kobe.warehouse.service.report;

import com.kobe.warehouse.service.dto.enumeration.TypeVenteDTO;
import com.kobe.warehouse.service.dto.report.DailySalesSummaryDTO;
import java.time.LocalDate;
import java.util.List;

/**
 * Service interface for sales summary reports using mv_daily_sales_summary
 */
public interface SalesSummaryReportService {
    /**
     * Get daily sales summary for a date range
     *
     * @param startDate start date (inclusive)
     * @param endDate end date (inclusive)
     * @return list of daily summaries
     */
    List<DailySalesSummaryDTO> getDailySalesSummary(LocalDate startDate, LocalDate endDate);

    /**
     * Get daily sales summary for a specific date
     *
     * @param date the date
     * @return list of summaries grouped by sale type
     */
    List<DailySalesSummaryDTO> getDailySalesSummaryByDate(LocalDate date);

    /**
     * Get daily sales summary filtered by sale type
     *
     * @param startDate start date (inclusive)
     * @param endDate end date (inclusive)
     * @param typeVente sale type (e.g., "VO", "VNO")
     * @return list of daily summaries for that type
     */
    List<DailySalesSummaryDTO> getDailySalesSummaryByType(LocalDate startDate, LocalDate endDate, TypeVenteDTO typeVente);
}
